compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Utils/AMDAIEDmaUtils.cpp

```diff
@@ -334,28 +334,50 @@ LogicalResult foldLinearDims(
 
   for (int i = offsets.size() - 2; i >= 0; i--) {
```

Comment:
```suggestion
  for (int i = static_cast<int>(offsets.size()) - 2; i >= 0; i--) {
```

Reply:
Fly by comment: I think the code is safe as is, but not obvious. `int i = (size_t)1 - 2` will produce compiler warnings (but `i` will be `-1`, so correct for the loop).